ui/scripts/generate-api-client.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+# Generate the API client for Vault UI using the OpenAPI spec
+# Vault Enterprise binary required to generate complete API client
+# Run 'make entdev' in vault-enterprise directory prior to running this script or an error will occur
+
+# exit if a command fails
+set -e
+
+# check if vault version contains +ent
+VERSION=$(vault version)
+if [[ ! "$VERSION" == *"+ent"* ]]; then
+  echo "Vault Enterprise binary not found and is required to generate the complete API client"
+  echo $VERSION
+  echo "Run 'make entdev' in vault-enterprise directory to build the binary and try again"
+  exit 1
+fi
+
+echo "Generating OpenAPI spec..."
+echo
+
+# directory where this script is located
+SOURCE_DIR=$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")
+# parent vault directory
+VAULT_DIR="$( cd -P "$SOURCE_DIR/../../" && pwd )"
+# scripts directory
+SCRIPTS_DIR="$VAULT_DIR/scripts"
+# api client directory
+API_CLIENT_DIR="$VAULT_DIR/ui/api-client"
```

Comment:
This could change to `ui/app/client` but I was playing around with including this as a local dependency and would prefer to get that working since it would make it easier if/when we pull it out into it's own repo.